scripts/lib/LinkChecker.ts

```diff
@@ -0,0 +1,128 @@
+// This code is a Qiskit project.
+//
+// (C) Copyright IBM 2023.
+//
+// This code is licensed under the Apache License, Version 2.0. You may
+// obtain a copy of this license in the LICENSE file in the root directory
+// of this source tree or at http://www.apache.org/licenses/LICENSE-2.0.
+//
+// Any modifications or derivative works of this code must retain this
+// copyright notice, and modified files need to carry a notice indicating
+// that they have been altered from the originals.
+
+import path from "node:path";
+
+const DOCS_ROOT = "./docs";
+const CONTENT_FILE_EXTENSIONS = [".md", ".mdx", ".ipynb"];
+
+export class File {
+  readonly path: string;
+  anchors: string[];
+
+  constructor(path: string) {
+    this.path = path;
+    this.anchors = [];
+  }
+
+  setAnchors(anchors: string[]) {
+    this.anchors = anchors;
+  }
```

Comment:
It'd be better for this to be set only in the constructor, and that `this.anchors` is readonly.